### PR TITLE
Handle button sheet destination without an inline sheet

### PR DESCRIPTION
**Pull request by MonikaMateska**
### Checklist
- [x] If applicable, unit tests
- [x] If applicable, create follow-up issues for `purchases-ios` and hybrids

## Motivation

While working on bringing Astra to our iOS app, I hit a draft paywall that refused to parse on iOS but rendered fine on the web dashboard, the same applies on Android as well.

A `navigate_to` action with `destination: "sheet"` can arrive without an inline `sheet` object (the backend types it as optional and drops null fields). The `ActionSerializer` required it via `checkNotNull(sheet)`, so a missing sheet
threw `IllegalStateException`. Because that runs during deserialization, one bad button took down the whole paywall parse. 
The web SDK types it as `sheet?: SheetProps | null` and renders a missing sheet as a button with a no-op tap.

## Description

This aligns Android with the web contract:

- `ButtonComponent.Destination.Sheet` fields are now optional (`stack` nullable,  `id` defaulted), so a content-less sheet is representable, and the deserializer  maps a missing sheet to an empty `Sheet` instead of `checkNotNull`-throwing.
- A `navigate_to`/`sheet` destination with no `stack` maps to a new `ButtonComponentStyle.Action.NoOp` in `StyleFactory`, the button renders but its tap does nothing, rather than being routed to `Unknown` (which hides the
  button). Sheets with content are unchanged.
- `NoOp` is handled across the action `when`s (`toPaywallAction` -> null tap, interaction mapping, purchase-related check), and `OfferingToStateMapper`  handles the now-nullable sheet stack.

Added a deserialization test covering a `sheet` destination with no inline `sheet` (decodes to an empty `Sheet` instead of throwing).

---

> [!NOTE]
> **Low Risk**
> Localized paywall JSON/UI behavior change for an edge-case button action; no purchase, auth, or billing flow changes.
> 
> **Overview**
> Fixes **whole-paywall parse failures** when a button uses `navigate_to` with `destination: "sheet"` but the optional inline `sheet` object is absent (payloads that already work on web).
> 
> **Deserialization:** `Destination.Sheet` now allows a nullable `stack` and defaults on other fields; `ActionSerializer` maps a missing inline sheet to an empty `Sheet()` instead of throwing during decode.
> 
> **Rendering:** `StyleFactory` turns sheet destinations with no `stack` into a new **`NoOp`** button action so the control still shows but taps do nothing (web parity). `NoOp` is wired through paywall action mapping, interaction analytics, and unsupported-condition walks; sheet destinations with content are unchanged.
> 
> Adds a unit test that `destination: "sheet"` without an inline `sheet` decodes successfully.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 915f625f86b6c125d931a1884173e20d3be32858. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/StyleFactory.kt [hunk 1]
```diff
@@ -823,18 +831,22 @@ internal class StyleFactory(
...
+                        .map { stackComponentStyle ->
+                            ButtonComponentStyle.Action.NavigateTo.Destination.Sheet(
+                                id = destination.id,
+                                name = destination.name,
+                                stack = stackComponentStyle,
+                                backgroundBlur = destination.backgroundBlur,
+                                size = destination.size,
+                            )
+                        }
+                } ?: Result.Success(null)
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/StyleFactory.kt]: Nitpick but could be good to log an error here where it should never actually arrive.

**Review by tonidero — APPROVED** [review]
Makes sense!

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/StyleFactory.kt (hunk 1 above)
Comment (MonikaMateska) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/StyleFactory.kt]: Good point, adding it now
